apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -48,14 +52,51 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
     const [searchParams, setSearchParams] = useSearchParams();
     const [query, setQuery] = useState('');
 
-    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
-
-    const [selectedModule, setSelectedModule] = useState(
-        !!paramModule && modulenames.includes(paramModule)
-            ? paramModule
-            : modulenames[0]
+    const getModule = (
+        searchParams: URLSearchParams,
+        setSearchParams: (
+            params: URLSearchParamsInit,
+            navigateOpts?: NavigateOptions
+        ) => void,
+        modulenames: string[]
+    ): string => {
+        const paramModule = searchParams.get('module') || null;
+
+        // A module has been specified and the package has this module
+        if (!!paramModule && modulenames.includes(paramModule)) {
+            return paramModule;
+        }
+
+        // A module has been specified and the package does not have this module
+        if (!!paramModule) {
+            const newSearchParams = new URLSearchParams(searchParams);
+            newSearchParams.delete('module');
+            setSearchParams(newSearchParams, { replace: true });
+        }
+
+        // The default is to return the first module in the list
+        return modulenames[0];
+    };
+
+    const selectedModule = getModule(
+        searchParams,
+        setSearchParams,
+        modulenames
     );
 
+    const setModule = (
+        searchParams: URLSearchParams,
+        setSearchParams: (
+            params: URLSearchParamsInit,
+            navigateOpts?: NavigateOptions
+        ) => void,
+        newModule: string
+    ): void => {
+        const newSearchParams = new URLSearchParams(searchParams);
+        newSearchParams.set('module', newModule);
+        setSearchParams(newSearchParams, { replace: true });
```

Comment:
This one probably shouldn't be `replace: true`, only the case where the module is not valid.